lib/ansible/plugins/lookup/__init__.py

```diff
@@ -101,3 +102,19 @@ def run(self, terms, variables=None, **kwargs):
             result_string = to_unicode(result_string)
         """
         pass
+
+    def find_needle(self, myvars, subdir, needle):
```

Comment:
to rephrase, we don't care that multiple files can match, it MUST return the first one it finds, that is why search path order matters.